inox2d-opengl/src/gl_buffer.rs

```diff
@@ -11,71 +9,50 @@ unsafe fn upload_array_to_gl<T>(gl: &glow::Context, array: &[T], target: u32, us
 	gl.buffer_data_u8_slice(target, bytes, usage);
 }
 
-unsafe fn reupload_array_to_gl<T>(gl: &glow::Context, array: &[T], target: u32, start_idx: usize, end_idx: usize) {
-	let slice = &array[start_idx..end_idx];
-	let bytes: &[u8] = core::slice::from_raw_parts(slice.as_ptr() as *const u8, core::mem::size_of_val(slice));
-	gl.buffer_sub_data_u8_slice(target, start_idx as i32, bytes);
-}
-
-pub trait RenderCtxOpenglExt {
-	unsafe fn setup_gl_buffers(
-		&self,
-		gl: &glow::Context,
-		vao: glow::VertexArray,
-	) -> Result<glow::VertexArray, OpenglRendererError>;
-	unsafe fn upload_deforms_to_gl(&self, gl: &glow::Context);
+/// Create a vertex array. Initialize vertex, uv, deform and index buffers, upload content and attach them to the vertex array. Return the array.
+///
+/// # Errors
+///
+/// This function will return an error if it couldn't create a vertex array.
+///
+/// # Safety
+///
+/// No prerequisites.
+pub unsafe fn setup_gl_buffers(
```

Comment:
If there are no prerequisites, I guess the function shouldn't be `unsafe` and it should just have an `unsafe` block inside.